[gas-profiler] add developer documentation

vgao1996 opened the pull request:
This adds the developer documentation for the updated gas profiler. It also amends some existing pages to remove duplicate contents and add links to the new doc.

vgao1996 commented on developer-docs-site/scripts/additional_dict.txt:
```diff
@@ -1,629 +1,608 @@
 ABI
 ABIs
+accessors
```
Sorted using the `sort-additional-dict.sh` script after adding a few words.

vgao1996 commented on developer-docs-site/static/gas-profiling/sample-report/index.html:
```diff
@@ -0,0 +1,1085 @@
+<!-- Copyright © Aptos Foundation -->
```
Generated file. No need to review its contents.

vgao1996 commented on developer-docs-site/docs/move/move-on-aptos/gas-profiling.md:
```diff
@@ -0,0 +1,129 @@
+---
```
This is the main documentation this PR is adding.

georgemitenkov commented on developer-docs-site/docs/concepts/gas-txn-fee.md:
```diff
@@ -133,7 +133,7 @@ We are aware of the confusion this might create, and plan to present these as se
 
 If a transaction deletes state items, a refund is issued to the transaction payer for the released storage slots. Currently, a full refund is issued for the slot's fee, excluding any fees for excess bytes beyond a set quota (e.g., 1KB). However, fees for event emissions are not refundable.
 
-The refund amount is denominated in APT and is not converted to gas units or included in the total `gas_used`. Instead, this refund amount is specifically detailed in the `storage_fee_refund_octas` field of the [`FeeStatement`](#the-fee-statement). As a result, the transaction's net effect on the payer's APT balance is determined by `gas_used * gas_unit_price - storage_refund`. If the result is p ...
```
nit: placing each sentence on each line helps a lot with makings diffs smaller and leaving comments

vgao1996 replied:
Well this is markdown, so if you put things are a new line, they'll be rendered as a separate paragraph.

georgemitenkov replied:
I am pretty sure you can place them on the next line, because the space is used for new paragraphs. Anyway,  this is my personal preference because it is way easier to refer to a specific sentence this way :)

vgao1996 replied:
Yeah I deleted my previous comment, realizing I was confused by myself.

I'll clean up the new doc I wrote and split things into lines, but probably won't fix other docs at this moment.

sausagee commented on developer-docs-site/docs/move/move-on-aptos/gas-profiling.md:
````diff
@@ -0,0 +1,129 @@
+$ cd aptos-move/move-examples/hello_blockchain
+```
+
+Then, we can simulate module publishing with the extra option `--profile-gas`. Notice that you do need to have your CLI profile set up properly and bind the named addresses correctly.
````
This seems like a common got-cha.  Can you make it in its own line and bold the words on "CLI profile set up"?

sausagee commented on developer-docs-site/docs/concepts/gas-txn-fee.md:
```diff
@@ -181,3 +181,6 @@ The simulation steps for finding the correct amount of gas for a transaction are
 :::
+
+## Gas Profiling
+The Aptos blockchain offers a convenient tool called the Aptos Gas Profiler that can help you understand the precise gas usage of your transaction through a web-based report. See [Gas Profiling](../move/move-on-aptos/gas-profiling) for more details.
```
```suggestion
The Aptos blockchain offers a powerful tool called the Aptos Gas Profiler that helps you understand the precise gas usage of your transaction through a web-based report. See [Gas Profiling](../move/move-on-aptos/gas-profiling) for more details. The gas profiler tool is integrated into the Aptos CLI. 
```

vgao1996 commented:
@georgemitenkov @sausagee I've addressed the comments from both of you. Please take another look! Thanks

davidiw commented on developer-docs-site/docs/concepts/gas-txn-fee.md:
```diff
@@ -181,3 +181,8 @@ The simulation steps for finding the correct amount of gas for a transaction are
 :::tip
 Prioritization is based upon buckets of `gas_unit_price`. The buckets are defined in [`mempool_config.rs`](https://github.com/aptos-labs/aptos-core/blob/30b385bf38d3dc8c4e8ee0ff045bc5d0d2f67a85/config/src/config/mempool_config.rs#L8). The current buckets are `[0, 150, 300, 500, 1000, 3000, 5000, 10000, 100000, 1000000]`. Therefore, a `gas_unit_price` of 150 and 299 would be prioritized nearly the ...
 :::
+
+## Gas Profiling
```
there's redundancy with the section on

`## Estimating gas consumption via simulation`

would be good to unify

davidiw commented on developer-docs-site/docs/move/move-on-aptos/gas-profiling.md:
````diff
@@ -0,0 +1,137 @@
+  - sources
+  - Move.toml
+```
+index.html is the main page of the report and you can simply open it in your web browser.
````
```suggestion
index.html is the main page of the report and you can open it in your web browser.
```

davidiw commented on developer-docs-site/docs/move/move-on-aptos/gas-profiling.md:
````diff
@@ -0,0 +1,137 @@
+```
+Again, it should be emphasized that even though the live chain-state is being used, this is still a local simulation so the module has NOT really been published to the target network.
+
+You can then find the generated gas report in a newly created directory called `gas-profiling`:
````
```suggestion
You can then find the generated gas report in the directory `gas-profiling`:
```

davidiw commented on developer-docs-site/docs/move/move-on-aptos/gas-profiling.md:
````diff
@@ -0,0 +1,137 @@
+  }
+}
+```
+Again, it should be emphasized that even though the live chain-state is being used, this is still a local simulation so the module has NOT really been published to the target network.
````
```suggestion
Again, it should be emphasized that even though the live chain-state is being used, this is a simulation so the module has NOT really been published to the target network.
```

davidiw commented on developer-docs-site/scripts/additional_dict.txt:
```diff
@@ -1,629 +1,609 @@
+accessors
+accountHasList
 ACK
+acount
```
acount? :roll_eyes: 
I see this wasn't you lol

davidiw reviewed (approved):
in the future, please separate out the changes to non-core components of a commit, the additional_dict update could should be in a separate commit.

georgemitenkov commented on developer-docs-site/docs/move/move-on-aptos/gas-profiling.md:
```diff
@@ -0,0 +1,137 @@
+## Future Plans
+We plan to extend the gas profiler with the following features:
+- Ability to replay historical transactions that have been committed (on mainnet, testnet etc.).
+- Ability to annotate source files
```
I think . is missing here

vgao1996 commented:
Thanks for the great feedback! I've addressed all issues.

> in the future, please separate out the changes to non-core components of a commit, the additional_dict update could should be in a separate commit.

Interesting. I touched this file only because of a failed lint. I've removed the sorting from this PR but still kept the few words I added, so that the tool won't complain about misspelled words in my docs. We can put up another PR to get the list sorted.
